src/Microsoft.ML.Api/TypedCursor.cs

```diff
@@ -26,6 +26,21 @@ public interface IRow<TRow> : IRow
         void FillValues(TRow row);
     }
 
+    /// <summary>
```

Comment:
>summary [](start = 9, length = 7)

could you outline the contract of that? 
If we get getters before the first `Accept`, what happens?

Also, `Accept` is a bit off, because it implies some choice on our part (accept or not). 
This mirrors 'FillValues', so maybe `ExtractValues`, or `ReadValues`?

Reply:
Hmmm yes.. let's do `Extract` since that's a nice counterpoint to `Fill`.

---
In reply to: [219579240](https://github.com/dotnet/machinelearning/pull/973#discussion_r219579240) [](ancestors = 219579240)

Reply:
Also good call I should have a check on `Position`.

---
In reply to: [219607464](https://github.com/dotnet/machinelearning/pull/973#discussion_r219607464) [](ancestors = 219607464,219579240)